nscq/src/functions.rs

```diff
@@ -0,0 +1,135 @@
+use std::ffi::{c_uint, c_void};
+
+use super::types::{
+    NscqCallback, NscqLabel, NscqObserver, NscqRc, NscqSession, NscqUuid, UserData,
+};
+
+fn error_to_str(rc: NscqRc) -> &'static str {
+    match rc {
+        0 => "NSCQ_RC_SUCCESS",
+        1 => "NSCQ_RC_WARNING_RDT_INIT_FAILURE",
+        -1 => "NSCQ_RC_ERROR_NOT_IMPLEMENTED",
+        -2 => "NSCQ_RC_ERROR_INVALID_UUID",
+        -3 => "NSCQ_RC_ERROR_RESOURCE_NOT_MOUNTABLE",
+        -4 => "NSCQ_RC_ERROR_OVERFLOW",
+        -5 => "NSCQ_RC_ERROR_UNEXPECTED_VALUE",
+        -6 => "NSCQ_RC_ERROR_UNSUPPORTED_DRV",
+        -7 => "NSCQ_RC_ERROR_DRV",
+        -8 => "NSCQ_RC_ERROR_TIMEOUT",
+        -127 => "NSCQ_RC_ERROR_EXT",
+        -128 => "NSCQ_RC_ERROR_UNSPECIFIED",
+        _ => "Unknown error code",
+    }
+}
+
+fn nsqc_handle_rc(rc: NscqRc) {
+    if rc < 0 {
+        panic!("NSCQ error: {}", error_to_str(rc));
```

Comment:
It would be much better to just encapsulate the error code into a proper `Result` with its own associated error type